lmdeploy/turbomind/deploy/source_model/llama.py

```diff
@@ -33,6 +43,29 @@ def __init__(self, new_params: dict, unused_params: dict, last_bin: bool, model_
         if tie_word_embeddings:
             self.output_weight_key = self.tok_embeddings_key
         self.processor = policy
+        if use_quant_online:
+            quant_params = self.quant_weight()
+            self.params.update(quant_params)
+
+    def quant_weight(self):
+        weight_cfg = self.model_cfg.get('weight_config')
+        assert weight_cfg['weight_type'] == 'fp8', (f"Unsupported weight_type: {weight_cfg['weight_type']}. "
+                                                    f'Expected fp8. ')
```

Comment:
we can reduce some redundancy here by removing the check.
The following `q_weight` can be updated by
```python
q_weight, scale = quant_blocked_fp8(weight, torch.float8_e4m3fn, block_size=128)
```
Since turbomind engine doesn't support other values of `block_size`